### PR TITLE
 Add `excludedPackages` option to the `prefer-add-event-listener` rule

### LINKED ISSUE #174: The `prefer-add-event-listener` rule should not apply to Koa's `app.context.onerror`

### PR DESCRIPTION
**Pull request by medusalix**
Resolves #174

### CODE REVIEW

File: rules/prefer-add-event-listener.js [hunk 1]
```diff
@@ -2,6 +2,8 @@
 const getDocsUrl = require('./utils/get-docs-url');
 const domEventsJson = require('./utils/dom-events.json');
 
+const excludedPackages = new Set(['koa', 'sax']);
```
Comment (futpib) [inline rules/prefer-add-event-listener.js]: Should we add an option to override or extend this set?

File: rules/prefer-add-event-listener.js [hunk 2]
```diff
@@ -72,11 +75,31 @@ const create = context => {
 			codePathInfo = codePathInfo.upper;
 		},
 
+		Program() {
+			isDisabled = false;
```
Comment (futpib) [inline rules/prefer-add-event-listener.js]: Why not just initialize the variable to `false`?

Comment (medusalix) [reply rules/prefer-add-event-listener.js]: Oh, I thought those variables were bound to the rule, so I wanted to reset `isDisabled` at the start of a new file 🤦‍♂️. That's what reviews are for 😄.

File: rules/prefer-add-event-listener.js (hunk 1 above)
Comment (medusalix) [reply rules/prefer-add-event-listener.js]: Yeah, we could add a configuration option, but I think we should keep these values as a default if no options are specified.

**Comment by sindresorhus** [comment]
@medusalix I think it should be an options-object, so we're able to add additional options in the future:

```js
'prefer-add-event-listener': ['error', {
	excludedPackages: ['foo']
}]
```

**Comment by sindresorhus** [comment]
It also needs to be documented.

**Comment by sindresorhus** [comment]
@futpib Looks good to you?

**Pull request merged**